### PR TITLE
prefix api routes

### PR DESCRIPTION
**Pull request by pngwn**
## Description

Please include a concise summary, in clear English, of the changes in this pull request. If it closes an issue, please mention it here.

Closes: #(issue)

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: gradio/routes.py [hunk 1]
```diff
@@ -257,14 +263,26 @@ def create_app(
         app_kwargs["lifespan"] = create_lifespan_handler(
             app_kwargs.get("lifespan", None), *delete_cache
         )
-        app = App(auth_dependency=auth_dependency, **app_kwargs)
+        app = App(auth_dependency=auth_dependency, **app_kwargs, debug=True)
+        router = APIRouter(prefix="/gradio_api")
```
Comment (freddyaboulton) [inline gradio/routes.py:267]: I think we should add a constant in `route_utils` and use this here and in the other files. Other than that, this approach of using the router lgtm!

**Comment by abidlabs** [comment]
@pngwn I fixed a few things on the Python side, backend looks good to me. I'll work on the fixing the tests shortly but just fyi looks like uploading is currently broken, because the upload url is not constructed correctly. If you can take a look at that, that'd be great, you can test with kitchen_sink or a simple demo:

```py
import gradio as gr

gr.Interface(lambda x:x, "image", "image").launch()
```

and try uploading a file, you'll get a broken UI:

image

image

File: gradio/processing_utils.py [hunk 2]
```diff
@@ -476,14 +476,14 @@ def _move_to_cache(d: dict):
...
+        url_prefix = f"{API_PREFIX}/stream/" if payload.is_stream else f"{API_PREFIX}/file="
         if block.proxy_url:
             proxy_url = block.proxy_url.rstrip("/")
-            url = f"gradio_api/proxy={proxy_url}{url_prefix}{payload.path}"
+            url = f"{API_PREFIX}/proxy={proxy_url}{url_prefix}{payload.path}"
         elif client_utils.is_http_url_like(payload.path) or payload.path.startswith(
             f"{url_prefix}"
         ):
-            url = payload.path
+            url = f"{payload.path}"
```
Comment (abidlabs) [inline gradio/processing_utils.py:486]: unnecessary

Comment (pngwn) [reply gradio/processing_utils.py:488]: Yep. Leftover from testing. Will remove.

**Pull request merged**